RemZiQ/nodejs-aws-be
# Task-4

**WHAT this PR do?**

- add SQL script for creating and filling tables
- refactor `getAllPRoducts` handler. Handler use data from DB
- add handler `addProduct`

**Additional scope**

- POST/products lambda functions returns error 400 status code if product data is invalid
- All lambdas return error 500 status code on any error (DB connection, any unhandled error in code)
- All lambdas do console.log for each incoming requests and their arguments
- Transaction based creation of product (in case stock creation is failed then related to this stock product is not created and not ready to

**How to test**

- check [SQL scipt](https://github.com/RemZiQ/nodejs-aws-be/pull/2/files#diff-c3ec6f74cdfdaac209707bee56ba45a830b0aaad4d95c8e079ee43d3a9a76b4c) 
- verify that getAllProducts work as expected [link](https://vzhywwsygf.execute-api.eu-west-1.amazonaws.com/dev/products)
- verify that u can add Product. Please note If you will miss the required fields in the body of request then you will receive `status: 400`. If all fields are presented then product will be added to 2 tables in DB. You can use Postman. [link](https://vzhywwsygf.execute-api.eu-west-1.amazonaws.com/dev/products)
- after successful calling `addProduct` endpoint with POST method please verify that product was added to DB via calling `getAllProducts`. The added product should be presented in the response.
- check  [front app](https://dngh4t03yzrs0.cloudfront.net/)

**Notes**

_ProductModel_

  title:                 string
  description:     string
  price:               number
  image:             string
  count:              number

**Screenshots**
_example of bad request Postman_

![image](https://user-images.githubusercontent.com/43277312/98520684-742b7f80-2283-11eb-8401-52f287b0e3e9.png)

_example of successful request Postman_

![image](https://user-images.githubusercontent.com/43277312/98521203-32e79f80-2284-11eb-822f-8ebe55980fe1.png)
